elisp/guix-devel.el

```diff
@@ -338,6 +338,46 @@ Each rule should have a form (SYMBOL VALUE).  See `put' for details."
   (add-before 'guix-devel-indent-modify-phases-keyword-2))
 
 
+;;; String inside sexp
+
+(require 'edit-indirect nil t)
+
+(defconst guix-devel-code-block-syntax
+  '("description" "synopsis"))
+
+(defun guix-devel-code-block-position ()
+  "Return begining and end of a string inside
+`guix-devel-code-block-syntax' sexp."
+  (save-excursion
+    (re-search-backward (rx-to-string
+                         `(and (syntax open-parenthesis)
+                               (or ,@guix-devel-code-block-syntax)
+                               not-wordchar)
+                         t))
+    (cons (re-search-forward (rx-to-string `(syntax string-quote) t))
+          (- (re-search-forward
+              (rx-to-string
+               `(and (syntax string-quote)
+                     (syntax close-parenthesis)
```

Comment:
This is unnecessary.

Reply:
Hm, I wouldn't rely on searching to define the end of the string.
Imagine, for example, the following description:

```
"This great package can do some impressive (even \"mind-blowing\")
stuff and you will not regret installing it."
```

If you try to edit it, it will be partially cut (at the end of the first
line).  So what about the following code instead?

```elisp
(cons (re-search-forward (rx (syntax string-quote)))
      (1- (progn (up-list 1 t) (point))))
```